middle_end/flambda2/types/meet_and_join_new.ml

```diff
@@ -1743,38 +1838,58 @@ and join_row_like :
           ~right_env:(Join_env.right_join_env join_env)
       in
       let other2 = join_case env other2 other2 in
-      Ok other2
-    | Ok other1, Ok other2 -> Ok (join_case join_env other1 other2)
+      Or_unknown.map other2 ~f:(fun other2 -> Or_bottom.Ok other2)
+    | Ok other1, Ok other2 ->
+      Or_unknown.map (join_case join_env other1 other2) ~f:(fun case ->
+          Or_bottom.Ok case)
   in
-  known, other
+  Or_unknown.map other ~f:(fun other -> known, other)
```

Comment:
Would everything be clearer/cleaner if unknown was on the other case rather than on everything ?

Reply:
So yes, everything would be clearer, but patching type_grammar for that change ends up too much work so I'm giving it up for now.